src/client/mod.rs

```diff
@@ -27,27 +29,34 @@ use tower_service::Service;
 /// types within `hyper::Client`.
 #[derive(Clone, Debug)]
 pub struct Client<C, B> {
-    inner: hyper::Client<C, B>,
+    inner: hyper::Client<C, LiftBody<B>>,
 }
 
-impl<C, B> Client<C, B> {
-    /// Create a new client from a `hyper::Client`
-    pub fn new(inner: hyper::Client<C, B>) -> Self {
+impl<B> Client<HttpConnector, B>
```

Comment:
this statement shoudl be ->

```rust
impl<C, B> Client<C, B> {
	pub fn new(client: Client<C, B>) -> Self { ... }
}
```

Reply:
I think it makes sense to have both. `new` being the simplest one which matches `hyper::Client::new()`, and then `with_hyper` or `with_client` to construct one with a particular builder/client.

Reply:
Im fine with that 👍 or even, a builder fn that just exposes a special builder?

Reply:
I've gone with `with_client` for now. The important bit is that the user returns a `hyper::Client<C, B>` where the `B` part is actually left to be inferred, since we actually require this to be `LiftBody<hyper::Body>`.